src/main/java/com/petstore/petstoreRest/controller/ExceptionHandlerController.java

```diff
@@ -0,0 +1,24 @@
+package com.petstore.petstoreRest.controller;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.springframework.dao.DataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+@ControllerAdvice
+public class ExceptionHandlerController {
+    @ExceptionHandler(RuntimeException.class)
```

Comment:
Here you are reacting on too mach generic exception and trying to handle them with if statements. But this is not the best solution.
1. Just imagine if you have tons of exceptions that you have to handle. How many if statements will you have?
2. every time you add new exception handling you have to touch/modify the existing code. This violates the SOLID open-close principle
3. there are no logging

Here's an example of how you can improve your logic:
```

import javax.persistence.EntityExistsException;
import javax.persistence.EntityNotFoundException;
import lombok.extern.slf4j.Slf4j;
import org.springframework.dao.DataIntegrityViolationException;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.lang.Nullable;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;
import org.springframework.web.context.request.ServletWebRequest;
import org.springframework.web.context.request.WebRequest;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;

@Slf4j
@RestControllerAdvice
public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {

    private static final HttpHeaders NO_HEADERS = new HttpHeaders();

    @ExceptionHandler(EntityNotFoundException.class)
    public ResponseEntity<?> handleEntityNotFoundException(
            EntityNotFoundException ex,
            WebRequest request) {
        return handleExceptionInternal(ex, ex.getMessage(), NO_HEADERS, HttpStatus.NOT_FOUND, request);
    }

    @ExceptionHandler(IllegalArgumentException.class)
    public ResponseEntity<?> handleIllegalArgumentException(
            IllegalArgumentException ex,
            WebRequest request) {
      log.warn("Api error occurred path={}, exception={}",
          ((ServletWebRequest) request).getRequest().getRequestURI(), ex.getMessage(),ex);
      return handleExceptionInternal(ex, ex.getMessage(), NO_HEADERS, HttpStatus.UNPROCESSABLE_ENTITY, request);
    }

    @ExceptionHandler(EntityExistsException.class)
    public ResponseEntity<?> handleEntityExistsException(
            EntityExistsException ex,
            WebRequest request) {
        return handleExceptionInternal(ex, ex.getMessage(), NO_HEADERS, HttpStatus.CONFLICT, request);
    }

    @ExceptionHandler(RuntimeException.class)
    public ResponseEntity<?> handleRuntimeException(
            RuntimeException ex,
            WebRequest request) {
        return handleExceptionInternal(ex, ex.getMessage(), NO_HEADERS, HttpStatus.BAD_REQUEST, request);
    }

    @ExceptionHandler(DataIntegrityViolationException.class)
    public ResponseEntity<?> handleDataIntegrityViolationException(
        DataIntegrityViolationException ex,
        WebRequest request) {
        return handleExceptionInternal(ex, ex.getMessage(), NO_HEADERS, HttpStatus.BAD_REQUEST, request);
    }

    @Override
    protected ResponseEntity<Object> handleExceptionInternal(
            Exception ex, @Nullable Object body, HttpHeaders headers,
            HttpStatus status, WebRequest request) {
        if (log.isDebugEnabled())
            log.debug("Api error occurred path={}, exception={}",
                    ((ServletWebRequest) request).getRequest().getRequestURI(), ex.getMessage(), ex);
        else if (status.is5xxServerError() || status.equals(HttpStatus.BAD_REQUEST)) {
            log.error("Api error occurred path={}, exception={}",
                    ((ServletWebRequest) request).getRequest().getRequestURI(), ex.getMessage(), ex);
        } else if (status.is4xxClientError()) {
            if (log.isErrorEnabled())
                log.error("Api error occurred path={}, exception={}",
                        ((ServletWebRequest) request).getRequest().getRequestURI(), ex.getMessage());
        }
        return super.handleExceptionInternal(ex, null, headers, status, request);
    }
}

```

Reply:
Changed